### DISCUSSION TITLE
How to get a zwave device status (Alive, Dead, Asleep, ..)

### DISCUSSION THREAD

**Discussion by mealgi** [zwave-js/zwave-js#6659] [General]
Hello,

I am currently using Z-WAve JS along with Domoticz version 2024.4.

My Z-Wave configuration includes around 25 devices mostly Wallplugs, Door Sensor, Energy meter from Fibaro and AEOTEC.

I would like to write a piece of code (python or JS) reading each Z-Wave device status and write the outcome of it into a Domoticz Device. The objective is to get notified when a change occurs mostly when a node is dead.

Is it possible to do it ? Do you have any example to share to help me acheiving this objective.

Many thanks for your reply.

BR

**Comment by AlCalzone** [comment]
I have no idea how Domoticz works - this sounds more like a question for their forums, assuming Domoticz lets you automate stuff?

In JS code, you can listen to the `dead` and `alive` events of nodes:
https://zwave-js.github.io/node-zwave-js/#/api/node?id=zwavenode-events

**Reply by mealgi** [reply]
Hello
Thank you for your reply.
I know how to address domoticz devices from Javascript or Python.
My question is more about how to listen do the events of nodes in JS.
Do you have examples I could reuse to design and build my solution.
BR
